### PR TITLE
Add file traits

### PR DESCRIPTION
**Pull request by sameo**
This is needed for supporting disk image formats like e.g. QCOW.

### CODE REVIEW

**Comment by liujing2** [comment]
@sameo I updated the patches removing MIT license. And solved the conflict for your PR. But it seems the website re-commit on your branch after conflict solving?

File: src/file_traits.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// Copyright 2018 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
```
Comment (liujing2) [inline src/file_traits.rs:3]: > // found in the LICENSE file.

 LICENSE change to LICENSE-BSD-3-Clause?

File: src/tempdir.rs [hunk 2]
```diff
@@ -0,0 +1,102 @@
+// Copyright 2017 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
```
Comment (liujing2) [inline src/tempdir.rs:3]: And here :)

File: src/seek_hole.rs [hunk 3]
```diff
@@ -0,0 +1,205 @@
+// Copyright 2018 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE.BSD file.
+//
+// SPDX-License-Identifier: BSD-3-Clause
```
Comment (liujing2) [inline src/seek_hole.rs:5]: Do we need add an Intel Corp with Apache license?

File: src/write_zeroes.rs [hunk 4]
```diff
@@ -0,0 +1,169 @@
+// Copyright 2018 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
```
Comment (liujing2) [inline src/write_zeroes.rs:4]: And here :)

**Comment by sameo** [comment]
@liujing2 All fixed now. I also replaced the travic CI with a buildkite one. Please let me know if it needs further work.

**Comment by sameo** [comment]
Closing that one, replacing it with #2

**Pull request closed without merging**